File: Python/stringsplitter.py
def string_splitter(st):
    mid = len(st)//2
    middle = st[mid]
    first = st[0:mid]
    second = st[mid+1:]
    return middle, first, second


if __name__ == "__main__":
    si = input("Enter an odd length string: ")
    m, f, s = string_splitter(si)
    print("Middle character: " + m)
    print("First half: " + f)
    print("Second half: " + s)

File: C++/stringsplitter.cpp
#include<iostream>
#include<string>
using namespace std;


int main() {
    string s, start, mid, end;
    int length;

    cout << "Enter an odd length string: ";
    getline(cin, s);

    length = s.length();
    start = s.substr(0, length/2);
    mid = s.substr(length/2, 1);
    end = s.substr(length/2 + 1, length/2);

    cout << "Middle character: " << mid << endl;
    cout << "First half: " << start << endl;
    cout << "Second half: " << end << endl;

    return 0;
}


